src/osd/PeeringState.cc

```diff
@@ -5369,6 +5516,7 @@ PeeringState::Recovering::react(const RequestBackfill &evt)
   if (!ps->async_recovery_targets.empty()) {
     pg_shard_t auth_log_shard;
     bool history_les_bound = false;
+    // FIXME: Uh-oh we have to check this return value; choose_acting can fail!
```

Comment:
Can it?  Our current acting set (and up set, presumably?) meets the requirements, so moving async recovery/backfill targets into acting should always work, right?

Reply:
You’re probably right; I didn’t evaluate these too carefully, it was more of a note to check. Will look more carefully at what’s going on (though if it’s not possible to fail shouldn’t we assert that?)